pkgs/development/python-modules/verilogae/default.nix

```diff
@@ -0,0 +1,100 @@
+{
+  buildPythonPackage,
+  fetchFromSourcehut,
+  stdenv,
+  llvmPackages,
+  rustc,
+  patchelf,
+  python,
+  setuptools-rust,
+  cargo,
+  clang,
+  rustPlatform,
+  lib,
+  zlib,
+  ncurses,
+  libxml2,
+  pkg-config,
+  pkgs,
+}:
+
+buildPythonPackage rec {
+  pname = "verilogae";
+  version = "0.9-beta-8";
+
+  src = fetchFromSourcehut {
+    owner = "~dspom";
+    repo = "OpenVAF";
+    rev = "b800153e84a8a640b07048d580fabafc25bbc841";
+    hash = "sha256-NKiJrXnxwghNYtKL4s3YjvQd/r3QWe5saCYp4N4aQ8w=";
+  };
+
+  format = "other";
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = "${src}/Cargo.lock";
+    outputHashes = {
+      "salsa-0.17.0-pre.2" = "sha256-6GssvV76lFr5OzAUekz2h6f82Tn7usz5E8MSZ5DmgJw=";
+    };
+  };
+
+  buildInputs = [
+    llvmPackages.libllvm
+    pkgs.libxml2
+    ncurses.dev
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    cargo
+    clang
+    rustPlatform.cargoSetupHook
+    rustc
+    llvmPackages.libllvm
+    llvmPackages.clang
+    setuptools-rust
+    libxml2
+    ncurses
+    zlib
+    patchelf
+    pkg-config
+  ];
+
+  configurePhase = "true";
+
+  buildPhase = ''
+    export CC="${llvmPackages.clang}/bin/clang"
+    export CXX="${llvmPackages.clang}/bin/clang++"
+
+    export CARGO_BUILD_RUSTFLAGS="-L${llvmPackages.libllvm}/lib -lLLVMCore -lLLVMCodeGen -lLLVMTarget -lLLVMAnalysis -lLLVMSupport -lLLVMInstCombine -lLLVMTransformUtils -lLLVMScalarOpts -lLLVMInstrumentation"
+
+    export LD_LIBRARY_PATH="${llvmPackages.libllvm}/lib:${pkgs.libxml2}/lib:${ncurses.dev}/lib:$LD_LIBRARY_PATH"
+    export PKG_CONFIG_PATH="${pkgs.libxml2}/lib/pkgconfig:${ncurses.dev}/lib/pkgconfig:$PKG_CONFIG_PATH"
```

Comment:
Are you trying to get something like `rustPlatform.bindgenHook` here? Try to replace with that (in `nativeBuildInputs`) instead.

Reply:
Aaah, thanks! Sorry for the late response I missed this. 

Rebuild but didn't help:
```

  = note: /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: /build/source/target/release/deps/verilogae-fb8d1d5dfe4d5b56.verilogae.143chio64zq3zlp5.rcgu.o.rcgu.o: in function `verilogae::back::CodegenCtx::gen_func_obj':
          143chio64zq3zlp5:(.text._ZN9verilogae4back10CodegenCtx12gen_func_obj17h2e24a8ab9bcf66c8E+0x24e9): undefined reference to `LLVMConstGEP'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: /build/source/target/release/deps/verilogae-fb8d1d5dfe4d5b56.verilogae.143chio64zq3zlp5.rcgu.o.rcgu.o: in function `std::sys::sync::once::futex::Once::call':
          143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x5e8): undefined reference to `LLVMGetGlobalPassRegistry'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x5f4): undefined reference to `LLVMInitializeCore'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x5fd): undefined reference to `LLVMInitializeCodeGen'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x606): undefined reference to `LLVMInitializeScalarOpts'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x60f): undefined reference to `LLVMInitializeVectorization'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x618): undefined reference to `LLVMInitializeIPO'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x621): undefined reference to `LLVMInitializeAnalysis'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x62a): undefined reference to `LLVMInitializeTransformUtils'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x633): undefined reference to `LLVMInitializeInstCombine'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x63c): undefined reference to `LLVMInitializeInstrumentation'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text.unlikely._ZN3std3sys4sync4once5futex4Once4call17hb7d82a99d59f928cE+0x645): undefined reference to `LLVMInitializeTarget'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: /build/source/target/release/deps/verilogae-fb8d1d5dfe4d5b56.verilogae.143chio64zq3zlp5.rcgu.o.rcgu.o: in function `mir_llvm::ModuleLlvm::optimize':
          143chio64zq3zlp5:(.text._ZN8mir_llvm10ModuleLlvm8optimize17h6e7431793b0b7dd4E+0x12): undefined reference to `LLVMPassManagerBuilderCreate'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text._ZN8mir_llvm10ModuleLlvm8optimize17h6e7431793b0b7dd4E+0x20): undefined reference to `LLVMPassManagerBuilderSetOptLevel'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text._ZN8mir_llvm10ModuleLlvm8optimize17h6e7431793b0b7dd4E+0x2b): undefined reference to `LLVMPassManagerBuilderSetSizeLevel'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text._ZN8mir_llvm10ModuleLlvm8optimize17h6e7431793b0b7dd4E+0x43): undefined reference to `LLVMPassManagerBuilderPopulateFunctionPassManager'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text._ZN8mir_llvm10ModuleLlvm8optimize17h6e7431793b0b7dd4E+0xc0): undefined reference to `LLVMPassManagerBuilderPopulateModulePassManager'
          /nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: 143chio64zq3zlp5:(.text._ZN8mir_llvm10ModuleLlvm8optimize17h6e7431793b0b7dd4E+0xea): undefined reference to `LLVMPassManagerBuilderDispose'
          collect2: error: ld returned 1 exit status
error: builder for '/nix/store/vwdzvc1f1pqlbv6xcnq0cg3z0v47r7ws-python3.11-verilogae-0.9-beta-8.drv' failed with exit code 101;
       last 10 log lines:
       >           collect2: error: ld returned 1 exit status
       >           
       >   = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
       >   = note: use the `-l` flag to specify native libraries to link
       >   = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib)
       >
       > error: could not compile `verilogae` (bin "verilogae") due to 1 previous error
       >
       > Caused by:
 ```